### ISSUE TITLE
read10xMatrix error

### ISSUE THREAD

**Issue by gengnomes** [kharchenkolab/pagoda2#127]
I've tried to read in 10x matrix folders from different chemistries and cellranger versions, but I always get this error:

#Reading in matrix...
#Reading in features...
#Error in setnames(ans, col.names) : 
#Can't assign 2 names to a 3 column data.table

I'm running R 4.1.0 and installed from the git repository using devtools. Would appreciate any help!

**Comment by evanbiederstedt** [comment]
Thanks for using pagoda2

Note that we have two functions here (for historical reasons):
`read.10x.matrices()` and `read10xMatrix()`

https://github.com/kharchenkolab/pagoda2/blob/master/R/helpers.R#L258-L319
https://github.com/kharchenkolab/pagoda2/blob/master/R/helpers.R#L403-L474

Could you give me a reproducible example so I could debug this? 

Please give me the command you are running. If possible, please should me an example dataset from 10x. 

Thanks, Evan

**Comment by gengnomes** [comment]
Thanks for the quick response! I was using read10xMatrix() which is referred to in the tutorial. The other command read.10x.matrices() worked.

**Comment by evanbiederstedt** [comment]
It might be useful to look at this issue as well: https://github.com/kharchenkolab/pagoda2/issues/116

This command works for me, as an example:

```
mydata = read10xMatrix("/Users/evanbiederstedt/Downloads/filtered_gene_bc_matrices2/mm10", version="V2")
```

> I was using read10xMatrix() which is referred to in the tutorial. 

Could you show me how you're using this function? We checked this on several 10x folders....I'm a bit surprised it's not working for you

**Comment by evanbiederstedt** [comment]
Actually, I do see a slight bug here in the newer 10x files. I can fix this. 

Thanks for pointing this out.

**Comment by evanbiederstedt** [comment]
Hi @gengnomes 

I fixed this behavior for version 1.0.6---it should be on CRAN soon. 

This works for me:
```
foo = read10xMatrix("/Users/evanbiederstedt/Downloads/filtered_feature_bc_matrix", version="V3")
```

I downloaded this file from https://www.10xgenomics.com/resources/datasets/10k-human-pbmcs-3-ht-v3-1-chromium-x-3-1-high

The issue was the following: there's now a third column for `features.tsv.gz`

Thanks for reporting this bug! 
Best, Evan
